crates/indexer/src/models/tokens.rs

```diff
@@ -145,20 +170,56 @@ impl Token {
                     ),
                     _ => (None, None, None),
                 };
-                if let Some((token, token_ownership)) = maybe_token_w_ownership {
+                if let Some((token, token_ownership, maybe_current_token_ownership)) =
+                    maybe_token_w_ownership
+                {
                     tokens.push(token);
                     token_ownerships.push(token_ownership);
+                    if let Some(current_token_ownership) = maybe_current_token_ownership {
+                        current_token_ownerships.insert(
+                            (
+                                current_token_ownership.token_data_id_hash.clone(),
+                                current_token_ownership.property_version.clone(),
+                                current_token_ownership.owner_address.clone(),
+                            ),
+                            current_token_ownership,
+                        );
+                    }
                 }
-                if let Some(token_data) = maybe_token_data {
+                if let Some((token_data, current_token_data)) = maybe_token_data {
                     token_datas.push(token_data);
+                    current_token_datas.insert(
+                        current_token_data.token_data_id_hash.clone(),
+                        current_token_data,
+                    );
                 }
-                if let Some(collection_data) = maybe_collection_data {
+                if let Some((collection_data, current_collection_data)) = maybe_collection_data {
                     collection_datas.push(collection_data);
+                    current_collection_datas.insert(
+                        current_collection_data.collection_data_id_hash.clone(),
+                        current_collection_data,
+                    );
                 }
             }
-            return (tokens, token_ownerships, token_datas, collection_datas);
+            return (
+                tokens,
+                token_ownerships,
+                token_datas,
+                collection_datas,
+                current_token_ownerships,
+                current_token_datas,
+                current_collection_datas,
+            );
         }
-        (vec![], vec![], vec![], vec![])
+        (
+            vec![],
+            vec![],
+            vec![],
+            vec![],
+            HashMap::default(),
+            HashMap::default(),
+            HashMap::default(),
+        )
```

Comment:
Btw, not sure, but I think you can get by with `Default::default()` and then get rid of the large tuple